nidx/nidx_paragraph/src/query_parser/stop_words.rs

```diff
@@ -90,6 +92,40 @@ pub fn is_stop_word(word: &str) -> bool {
     STOP_WORDS.is_stop_word(word)
 }
 
+/// Consume a tokenized query and filter out any stop words from it.
+///
+/// A stop words is any literal matching the following criteria:
+/// - It is present in the given list of stop words
+/// - Is **not** the last term in the query
+///
+/// The last term of the query can be used as fuzzy prefix and must be preserved.
+pub fn remove_stop_words(mut query: Vec<Token>) -> Vec<Token> {
+    if query.is_empty() {
+        return query;
+    }
+    let query_len = query.len();
+
+    // as we don't want to remove the last literal, we pop the last element and
+    // always put it in the filtered query. We don't really care if the last
+    // it's a literal or not.
+    let last: Token<'_> = query.pop().unwrap(); // safe as query is not empty
+
+    query
+        .into_iter()
+        .take(query_len - 1)
```

Comment:
I think this `take` should not be here? `query.pop()` has already removed the last element, so this skips over the penultimate element.